### PR TITLE
Init Container uses same value of resources as kafka resources

### PR DESCRIPTION
**Pull request by ShubhamRwt**
### Type of change

_Select the type of your PR_

- Enhancement / new feature

### Description

This PR resolves the issue mentioned in #5372. This PR makes the init Container resources configurable in such a way that the value of resources which were configured for kafka resources are used for setting the resources of the Init Containter also.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java [hunk 1]
```diff
@@ -1548,24 +1545,29 @@ protected Affinity getMergedAffinity() {
...
+            } else {
+                initContainer = new ContainerBuilder()
+                        .withName(INIT_NAME)
+                        .withImage(initImage)
+                        .withArgs("/opt/strimzi/bin/kafka_init_run.sh")
+                        .withEnv(getInitContainerEnvVars())
+                        .withVolumeMounts(VolumeUtils.createVolumeMount(INIT_VOLUME_NAME, INIT_VOLUME_MOUNT))
+                        .withImagePullPolicy(determineImagePullPolicy(imagePullPolicy, initImage))
+                        .withSecurityContext(templateInitContainerSecurityContext)
+                        .build();
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java:1569]: This seems like it should work fine. But I think it duplicates too much code. Why don't you create the init container first and only then set the resources ... not a complete code, but something like this:

```
Container initContainer = new ContainerBuilder().......build();

if (getResources() != null) {
    initContainer.setResources(getResources());
}
```

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java [hunk 2]
```diff
@@ -531,30 +529,37 @@ public Deployment generateDeployment(Map<String, String> annotations, boolean is
...
+                initContainer = new ContainerBuilder()
+                        .withName(INIT_NAME)
+                        .withImage(initImage)
+                        .withArgs("/opt/strimzi/bin/kafka_init_run.sh")
+                        .withEnv(getInitContainerEnvVars())
+                        .withVolumeMounts(VolumeUtils.createVolumeMount(INIT_VOLUME_NAME, INIT_VOLUME_MOUNT))
+                        .withImagePullPolicy(determineImagePullPolicy(imagePullPolicy, initImage))
+                        .withSecurityContext(templateInitContainerSecurityContext)
+                        .build();
+            }
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java:554]: Same as above.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java [hunk 3]
```diff
@@ -531,30 +529,37 @@ public Deployment generateDeployment(Map<String, String> annotations, boolean is
...
     }
 
     private ResourceRequirements getInitContainerResourceResourceRequirements() {
-        return new ResourceRequirementsBuilder()
-                .addToRequests("cpu", new Quantity("100m"))
-                .addToRequests("memory", new Quantity("128Mi"))
-                .addToLimits("cpu", new Quantity("1"))
-                .addToLimits("memory", new Quantity("256Mi"))
-                .build();
+        return getResources();
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java:562]: Is this `getInitContainerResourceResourceRequirements` method used also somewhere else? If not, do we really need it? Why not call `getResources()` directly from `getInitContainers()`? It seems not needed if it is not called from anywhere else.

Comment (ShubhamRwt) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java:562]: Yeah, that was my initial thought. I will check if it is being used somewhere and if not, I will exclude it.

Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java:562]: Even if it is used somewhere else, you should think if it makes sense to have a separate method for it or not.

Comment (ShubhamRwt) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java:562]: It is not being used at any other place

**Review by scholzj — COMMENTED** [review]
The code looks good. But checkstyle is failing, so you will need to have a look at it.

**Comment by ShubhamRwt** [comment]
Resolved @scholzj

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Review by sknot-rh — APPROVED** [review]

**Pull request merged**